tools/parseAndPopulate/modules.py

```diff
@@ -212,13 +216,32 @@ def get_json(js):
             self.__resolve_contact()
             self.__resolve_description()
             self.__resolve_document_name_and_reference(document_name, reference)
+            self.__resolve_tree()
+            self.__resolve_expiration()
             self.__resolve_module_classification(module_classification)
             self.__resolve_working_group()
             self.__resolve_author_email(author_email)
             self.__resolve_maturity_level(maturity_level)
             self.__resolve_semver()
             self.__resolve_tree_type()
 
+    def __resolve_tree(self):
+        if self.module_type == 'module':
+            self.tree = 'services/tree/{}@{}.yang'.format(self.name,
+                                                          self.revision)
+
+    def __resolve_expiration(self):
+        if self.reference and 'datatracker.ietf.org' in self.reference:
+            ref = self.reference.split('/')[-1]
+            url = ('https://datatracker.ietf.org/api/v1/doc/document/'
+                   + ref + '/?format=json')
+            response = requests.get(url)
+            if response.status_code == 200:
+                data = json.loads(response.content)
+                if '/api/v1/doc/state/2/' in data['states']:
+                    self.expired = True
+                    self.expiration_date = data['expires']
+
     def __save_file(self, to):
```

Comment:
This function looks like it should be mainly data-driven, rather than lots of "ifs". Can we refactor separately from this commit? Same per-release update issue from above. Seems that all the data driving could happen by understanding the directory structure in GitHub:

```
.
├── nx
│   ├── 7.0-3-F1-1
│   ├── 7.0-3-F2-1
│   ├── 7.0-3-F2-2
│   ├── 7.0-3-F3-1
│   ├── 7.0-3-I5-1
│   ├── 7.0-3-I5-2
│   ├── 7.0-3-I6-1
│   ├── 7.0-3-I6-2
│   ├── 7.0-3-I7-1
│   └── 7.0-3-I7-2
├── xe
│   ├── 1631
│   │   └── MIBS
│   ├── 1632
│   │   └── MIBS
│   ├── 1641
│   │   └── MIBS
│   ├── 1651
│   │   └── MIBS
│   ├── 1661
│   │   └── MIBS
│   ├── 1662
│   │   └── MIBS
│   └── 1671
│       └── MIBS
└── xr
    ├── 530
    ├── 531
    ├── 532
    ├── 533
    ├── 534
    ├── 600
    ├── 601
    ├── 602
    ├── 611
    ├── 612
    ├── 613
    ├── 621
    ├── 622
    └── 631
```